state-chain/amm/src/limit_orders.rs

```diff
@@ -245,15 +241,15 @@ impl SwapDirection for BaseToQuote {
 }
 impl SwapDirection for QuoteToBase {
 	fn input_amount_ceil(output: Amount, price: Price) -> Amount {
-		mul_div_ceil(output, price, U256::one() << PRICE_FRACTIONAL_BITS)
+		price.output_amount_ceil(output)
 	}
 
 	fn input_amount_floor(output: Amount, price: Price) -> Amount {
 		BaseToQuote::output_amount_floor(output, price)
 	}
 
 	fn output_amount_floor(input: Amount, price: Price) -> Amount {
-		mul_div_floor(input, U256::one() << PRICE_FRACTIONAL_BITS, price)
+		price.input_amount_floor(input)
```

Comment:
Maybe worth leaving a comment to say why we calculate output with a funciton with "input" in it.